### ISSUE TITLE
v0.1.0b2 requests pyzmq 

### ISSUE THREAD

**Issue by laynr** [jupyterlite/jupyterlite#551]
I am noticing after installing v0.1.0b2 from v0.1.0b0 (maybe it started in v0.1.0b1), that there is a request for `pyzmq` (https://pypi.org/pypi/pyzmq/json).  There is not a *py3-none-any.whl version of `pyzmq` at https://pypi.org/project/pyzmq/#files.  

In offline  builds I put all required  whls files in ./pypi  at build time.  This allows them to be pulled locally instead of externally.  What whl can I put in my local pypi to satisfy the request for `pyzmq`?

**Comment by bollwyvl** [comment]
> there is a request

This request occurs:
- when downloading wheels?
- when trying to _start_ a kernel?
- when you try to `piplite.install` something _other_ than the kernel?

As part of the baseline install to support `pyolite`, we ship a few [shim packages](https://github.com/jupyterlite/jupyterlite/blob/v0.1.0b2/packages/pyolite-kernel/src/_pypi.ts): unless you _really_ know you want to replace them (e.g. you customized them further), you'll want to make sure none of those wheel name prefixes appear in _your_ pypi... i'm wagering it's `ipykernel`, but it's very hard to be sure without any kind of log message.

**Comment by laynr** [comment]
So your saying don't download fresh `ipykernel`, `piplite`, or `widgetsnbextension` whl files and put them the pypi folder?   I can try that.

The request occurs after going to /lab/index.html and then clicking pyolite under notebooks the error is visible in the developer console. It fails to fetch the file and then gets a "uncaught (in promise) python error".

As I said I am trying to do this offline. So I:
- put the top 5000 downloaded python package names in my `requirements.txt` 
- ran ` cat requirements.txt | xargs -n 1 python3.9 -m pip download -d ./pypi --no-deps --only-binary :all: --platform none`
- put all those 1000's of  whl files in `pypi` folder and build with  `jupter lite build -- pyodide ./pyodide-build-0.19.1tar.bz2 --output-dir ./public and then go offile.
- oh and I had previously installed with ` python3.9 -m pip install --user ./jupyterlite-0.1.0b2-py3-none-any.whl[all]  --index-url=http<innteral pypi/simple> --tristed-host=<internal>-host  

Note: I wish I could just modified `jupyter-lite.json` to point to the the internal pypi server, but it is just a `simple` server, it does not recognize ../package/json requests.  **Is there any plan to support simple pypi mirrors in the future?**

**Comment by bollwyvl** [comment]
> So your saying don't download fresh ipykernel, piplite, or
> widgetsnbextension whl files and put them the pypi folder? I can try that.
>
Right. Don't overload anything shipped with pyolite unless you really mean
it.

>  *Is there any plan to support simple pypi mirrors in the future?*
>
This would be an upstream feature request or PR for:

https://github.com/pyodide/pyodide

But: PRs may not be welcome to do that.

Pyolite/piplite added the ability to load custom wheel indexes, but didn't
change how the parsing works.

This was mostly to avoid having to hard-code the kernel dependencies, and
offer a way to cache a known, small set of (custom) wheels.

I think with /simple it's a much harder problem, as the index doesn't have
enough information for a solve without looking inside packages.
